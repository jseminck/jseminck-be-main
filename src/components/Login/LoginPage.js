import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from './loginActions';
import LoginInput from './LoginInput';

import './LoginPage.scss';

export class LoginPage extends React.Component {
  static propTypes = {
      login: PropTypes.object.isRequired,
      actions: PropTypes.object.isRequired
  }

  render() {
    return (
      <div className="login">
        <fieldset>
          <legend className="legend">Login</legend>

          <LoginInput
            type="input"
            placeholder="E-mail"
            icon="fa-envelope-o"
            onChange={this.props.actions.onChangeEmail}
            value={this.props.login.email}
          />

          <LoginInput
            type="password"
            placeholder="Password"
            icon="fa-lock"
            onChange={this.props.actions.onChangePassword}
            value={this.props.login.password}
          />

          <button type="submit" className="submit">
            <i className="fa fa-long-arrow-right"></i>
          </button>

        </fieldset>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {login: state.login};
}

function mapDispatchToProps(dispatch) {
  return {actions: bindActionCreators(actions, dispatch)};
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);

