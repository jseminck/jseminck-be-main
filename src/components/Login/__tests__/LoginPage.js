import React from "react";
import renderer from 'react-test-renderer';
import {LoginPage} from "./../LoginPage";

describe("LoginPage", function() {
    it('renders correctly', () => {
        const app = renderer.create(
          <LoginPage
              login={{
                login: 'myUser',
                password: 'myPassword'
              }}
              actions={{
                onChangeEmail: () => {}
              }}
          />
        ).toJSON();

        expect(app).toMatchSnapshot();
    });
});