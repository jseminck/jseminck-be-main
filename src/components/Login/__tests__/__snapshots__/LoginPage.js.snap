exports[`LoginPage renders correctly 1`] = `
<div
  className="login">
  <fieldset>
    <legend
      className="legend">
      Login
    </legend>
    <div
      className="input">
      <input
        onChange={[Function onChangeEmail]}
        placeholder="E-mail"
        required={true}
        type="input"
        value={undefined} />
      <span>
        <i
          className="fa fa-envelope-o" />
      </span>
    </div>
    <div
      className="input">
      <input
        onChange={undefined}
        placeholder="Password"
        required={true}
        type="password"
        value="myPassword" />
      <span>
        <i
          className="fa fa-lock" />
      </span>
    </div>
    <button
      className="submit"
      type="submit">
      <i
        className="fa fa-long-arrow-right" />
    </button>
  </fieldset>
</div>
`;
